rmw_cyclonedds_cpp/src/rmw_node.cpp

```diff
@@ -2011,9 +2024,64 @@ static rmw_time_t dds_duration_to_rmw(dds_duration_t duration)
   }
 }
 
+static rosidl_type_hash_t get_type_hash(const rosidl_message_type_support_t * type_support)
+{
+  const auto * id = type_support->typesupport_identifier;
```

Comment:
This is my personal bugaboo, but can we use the actual type here (rather than `auto`)?  Otherwise I don't know what this thing is.